rmw_fastrtps_dynamic_cpp/src/rmw_node.cpp

```diff
@@ -72,6 +81,13 @@ rmw_create_node(
 rmw_ret_t
 rmw_destroy_node(rmw_node_t * node)
 {
+  RMW_CHECK_ARGUMENT_FOR_NULL(node, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    node,
+    node->implementation_identifier,
+    eprosima_fastrtps_identifier,
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
+  assert(node->context != nullptr);
```

Comment:
Same argument as above; this should check and return an error, I think.

Reply:
Same as above.